landoapi/stacks.py

```diff
@@ -169,10 +169,18 @@ def block(node, reason):
     # We won't land anything that has a repository we don't support, so make
     # a pass over all the revisions and block these.
     for phid, revision in revision_data.revisions.items():
-        if (
-            PhabricatorClient.expect(revision, "fields", "repositoryPHID")
-            not in landable_repos
-        ):
+        repo = PhabricatorClient.expect(revision, "fields", "repositoryPHID")
+        if not repo:
+            block(
+                phid,
+                "No repository configured for this revision. Please "
+                "specify a repository for this revision using the "
+                "'Edit revision' option in the top-right of the "
+                "Phabricator UI.",
```

Comment:
These messages end up in a small tooltip, so keeping it a little less verbose would probably be preferrable. How about something like: `'Revision\'s repository unset. Specify a target using "Edit revision" in Phabricator'`